pkg/controller/wip/controller_test.go

```diff
@@ -138,6 +165,125 @@ func TestReconcileBroker(t *testing.T) {
 
 }
 
+func TestReconcileBrokerErrorFetchingCatalog(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerCatalog, _, _, testController, _, stopCh := newTestController(t)
+	defer close(stopCh)
+
+	fakeBrokerCatalog.RetErr = fakebrokerapi.ErrInstanceNotFound
+	broker := getTestBroker()
+
+	testController.reconcileBroker(broker)
+
+	actions := filterActions(fakeCatalogClient.Actions())
+	if e, a := 1, len(actions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v. Actions: %+v", e, a, actions)
+	}
+	updateAction := actions[0].(core.UpdateAction)
+	if e, a := "update", updateAction.GetVerb(); e != a {
+		t.Fatalf("Unexpected verb on actions[1]; expected %v, got %v", e, a)
+	}
+	updateObject := updateAction.GetObject().(*v1alpha1.Broker)
+	if e, a := broker.Name, updateObject.Name; e != a {
+		t.Fatalf("Unexpected name of broker: expected %v, got %v", e, a)
+	}
+	if e, a := v1alpha1.ConditionFalse, updateObject.Status.Conditions[0].Status; e != a {
+		t.Fatalf("Unexpected condition status: expected %v, got %v", e, a)
+	}
+
+	kubeActions := fakeKubeClient.Actions()
+	if e, a := 0, len(kubeActions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
+	}
+}
+
+func TestReconcileBrokerWithAuthError(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, _, _, _, testController, _, stopCh := newTestController(t)
+	defer close(stopCh)
+
+	broker := getTestBroker()
+	broker.Spec.AuthSecret = &v1.ObjectReference{
```

Comment:
There's a test for that already -- TestReconcileBroker.  But, it's not an error right now, but I'm happy to make it an error if we can point to a place where the spec says it's required.